indigo_app/views/documents.py

```diff
@@ -139,11 +138,41 @@ def get(self, request, *args, **kwargs):
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
         context['provision_eid'] = self.eid
+        context['provision_counters'], context['eid_counter'] = self.get_counters()
         return context
 
     def get_document_content_json(self, document):
         return json.dumps(etree.tostring(self.provision_xml, encoding='unicode'))
 
+    def get_counters(self):
+        """ Generate provision and eid counters based on the content preceding our element.
```

Comment:
nice - as I understand it, you're chopping off everything after this element in the tree. So we'll generate eIds for everything up to it (excluding it).

Reply:
exactly, yes, which is what the rewriter uses when calculating an eid / checking it for uniqueness